components/General/TWModal/index.tsx

```diff
@@ -35,6 +35,11 @@ export const TWModal: React.FC<TWModalProps> = ({ open, onClose, size = 'default
                         <Dialog.Overlay className="fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity" />
                     </Transition.Child>
 
+                    {/* This element is to trick the browser into centering the modal contents. */}
+                    <span className="inline-block h-screen align-middle" aria-hidden="true">
+                        &#8203;
+                    </span>
+
```

Comment:
whould be interested to know how this works if you could explain

Reply:
This is the recommended implementation from headless ui docs. We had this code implemented previously but I didn't fully understand what it was doing and the idea of tricking the browser kept bugging me so I replaced with traditional css modal positioning while working on the commit modal.

My implementation led to some overlay positioning bugs on screens smaller than the modal height.

I did spend time trying to solve the initial issue just with css... which led me to stackoverflow... then github issues.. finally back to their docs and realize it was their recommendation.